api/v1alpha2/zz_generated.conversion.go

```diff
@@ -1009,6 +1009,7 @@ func autoConvert_v1alpha3_Instance_To_v1alpha2_Instance(in *v1alpha3.Instance, o
 	// WARNING: in.RootVolume requires manual conversion: does not exist in peer-type
 	out.NetworkInterfaces = *(*[]string)(unsafe.Pointer(&in.NetworkInterfaces))
 	out.Tags = *(*map[string]string)(unsafe.Pointer(&in.Tags))
+	// WARNING: in.AvailabilityZone requires manual conversion: does not exist in peer-type
```

Comment:
To preserve this value through a conversion that round-trips through v1alpha2, this will also need to restore the value from annotation (if present) similar to what is done around FailureDomain handling here: https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/d3d8d2f2b505e090d0bfe517c378defb8601cffc/api/v1alpha2/awscluster_conversion.go#L62

Reply:
I might be wrong but it seems to me that this case is already handled here https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/d3d8d2f2b505e090d0bfe517c378defb8601cffc/api/v1alpha2/awscluster_conversion.go#L66 since `Instance` is wrapped inside `AwsCluster` .

Reply:
Ah, indeed it is.